File: BAEKJOON/Golang/11000/11500/11557.go
package main

import "fmt"

func main() {
	var t int
	fmt.Scanf("%d", &t)

	for i := 0; i < t; i++ {
		var n int
		fmt.Scanf("%d", &n)
		var name string
		var maxAlcohol int
		maxAlcohol = -1
		for j := 0; j < n; j++ {
			var n string
			var a int
			fmt.Scanf("%s %d", &n, &a)
			if maxAlcohol < a {
				name, maxAlcohol = n, a
			}
		}
		fmt.Println(name)
	}
}

File: BAEKJOON/Python/11000/11500/11557.py
"""
Title : Yangjogang of The Year
Link : https://www.acmicpc.net/problem/11557
"""

from sys import stdin

input = stdin.readline
II = lambda: int(input())


if __name__ == "__main__":
    for _ in range(II()):
        name, max_alcohol = "", -1
        for _ in range(II()):
            n, a = input().strip().split()
            a = int(a)
            if max_alcohol < a:
                name, max_alcohol = n, a
        print(name)
